examples/js/utils/BufferGeometryUtils.js

```diff
@@ -381,6 +381,185 @@ THREE.BufferGeometryUtils = {
 
 		return new THREE.BufferAttribute( array, itemSize, normalized );
 
+	},
+
+	/**
+	 * Checks for duplicate vertices with hashmap.
+	 * Duplicated vertices are removed
+	 * Returns a new indexed THREE.BufferGeometry.
+	 * @param {THREE.BufferGeometry} geometry The geometry to merge its vertices.
+	 * @param {Integer} elementSize Use: 1 for points, 2 for line segments, 3 for triangles.
+	 * @return {THREE.BufferGeometry}
+	 */
+
+	mergeVertices: function ( geometry, elementSize ) {
+
+		if ( geometry.attributes.position === undefined ) return null;
+
+		var positions = geometry.attributes.position.array;
+
+		var numOriginalVerts = Math.floor( positions.length / 3 );
+
+		// Hashmap for looking up vertices by position coordinates (and making sure they are unique)
+		var verticesMap = {};
+
+		var numUnique = 0;
+		var changes = [];
+
+		var precisionPoints = 4; // number of decimal points, e.g. 4 for epsilon of 0.0001
+		var precision = Math.pow( 10, precisionPoints );
+
+		var pos = new THREE.Vector3();
+
+		for ( var i = 0; i < numOriginalVerts; i ++ ) {
+
+			var i3 = i * 3;
+			pos.set( positions[ i3 ], positions[ i3 + 1 ], positions[ i3 + 2 ] );
+			var vertKey = Math.round( pos.x * precision ) + '_' + Math.round( pos.y * precision ) + '_' + Math.round( pos.z * precision );
+
+			if ( verticesMap[ vertKey ] === undefined ) {
+
+				verticesMap[ vertKey ] = i;
+				changes[ i ] = numUnique ++;
+
+			} else {
+
+				//console.log('Duplicate vertex found. ', i, ' could be using ', verticesMap[vertKey]);
+				changes[ i ] = changes[ verticesMap[ vertKey ] ];
+
+			}
+
+		}
+
+		// Construct new indices, discarding elements with some collapsed vertices
+
+		var isIndexed = geometry.index !== null;
+
+		var vertexIndexFunc = null;
+
+		var numOriginalIndices = 0;
+
+		if ( isIndexed ) {
+
+			var indices1 = geometry.index.array;
+
+			vertexIndexFunc = function ( idx ) {
+
+				return indices1[ idx ];
+
+			}
+
+			numOriginalIndices = indices1.length;
+
+		}
+		else {
```

Comment:
nit: If there is an else statement, it should be on the same line as the closing bracket of the block, like `} else {`.

https://github.com/mrdoob/three.js/wiki/Mr.doob's-Code-Style%E2%84%A2#if-statement